src/main/java/seedu/address/storage/JsonAdaptedModuleClass.java

```diff
@@ -45,30 +53,77 @@ public JsonAdaptedModuleClass(ModuleClass source) {
         studentUuids.addAll(source.getStudentUuids().stream()
                .map(JsonAdaptedUuid::new)
                .collect(Collectors.toList()));
+        lessons.addAll(source.getLessons().stream()
+                .map(JsonAdaptedLesson::new)
+                .collect(Collectors.toList()));
     }
 
     /**
-     * Converts this Jackson-friendly adapted class object into the model's {@code ModuleClass} object.
+     * Gets a {@code List} of {@code UUID}s from {@code studentUuids}.
      *
-     * @throws IllegalValueException if there were any data constraints violated in the adapted class.
+     * @throws IllegalValueException if any of the {@code UUID}s are null or invalid.
      */
-    public ModuleClass toModelType() throws IllegalValueException {
-        final List<UUID> moduleClassUuids = new ArrayList<>();
+    private List<UUID> getUuidList() throws IllegalValueException {
+        List<UUID> uuidList = new ArrayList<>();
         for (JsonAdaptedUuid studentUuid : studentUuids) {
             if (studentUuid == null) {
-                throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, "studentUuid"));
+                throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, STUDENT_UUID_FIELD));
             }
 
             // catch invalid UUID
             try {
                 String uuidString = studentUuid.getUuidString();
                 UUID.fromString(uuidString);
             } catch (IllegalArgumentException e) {
-                throw new IllegalValueException(String.format(INVALID_FIELD_MESSAGE_FORMAT, "studentUuid"));
+                throw new IllegalValueException(String.format(INVALID_FIELD_MESSAGE_FORMAT, STUDENT_UUID_FIELD));
+            }
+            uuidList.add(studentUuid.toModelType());
+        }
+        return uuidList;
+    }
+
+    /**
+     * Checks if a {@code lesson} already exists in a list.
+     * Duplicates are detected by calling {@code isSameLesson} method in {@code Lesson}.
+     * Returns true if there is a duplicate.
+     */
+    private boolean hasDuplicateLessons(List<Lesson> lessons, Lesson lessonToCheck) {
```

Comment:
Perhaps we can simplify this method to:
```suggestion
    private boolean hasDuplicateLessons(List<Lesson> lessons, Lesson lessonToCheck) {
        return lessons.stream().anyMatch(lessonToCheck::isSameLesson);
    }
```